src/main/java/io/cdap/delta/bigquery/BigQueryUtils.java

```diff
@@ -60,9 +60,8 @@ private BigQueryUtils() {
    * missing '_sequence_num' column in any of the tables, exception will be thrown.
    */
   static long getMaximumExistingSequenceNumber(Set<SourceTable> allTables, String project, @Nullable String datasetName,
```

Comment:
Logic seems little bit complicated. We already have method that gives max sequence number from Set of tables (`allTables`), so we just have to make sure that this set will have max 1000 tables. something similar to below

```java
    Set<SourceTable> allRemainingTables = context.getAllTables();
    long maxExisting = 0;
    while (allRemainingTables.size() > 0) {
     // select 1000 tables from all tables
      HashSet<SourceTable> currentBatch = Sets.newHashSet(Iterables.limit(allRemainingTables, 1000));
     // find max among those 1000 tables
      long currentMax = BigQueryUtils.getMaximumExistingSequenceNumber(currentBatch, project, 
                                                                       conf.getDatasetName(), bigQuery, 
                                                                       encryptionConfig);
      maxExisting = Math.max(maxExisting, currentMax);
     // remove current batch of tables from all tables.
      allRemainingTables = Sets.difference(allRemainingTables, currentBatch);
    }
    
    return maxExisting;
```

Reply:
Added an extra method to handle this logic